lib/simple_saml/assertion.ex

```diff
@@ -187,4 +213,38 @@ defmodule SimpleSaml.Assertion do
         {:error, {:failed_to_parse_timestamp, iso8601_datetime}}
     end
   end
+
+  defp extract_attributes(node) do
+    with {:ok, attributes_node} <- XmlNode.first_child(node, "*:AttributeStatement") do
+      XmlNode.children(attributes_node, "*:Attribute")
+      |> Enum.flat_map(fn attribute_node ->
+        with {:ok, name} <- XmlNode.attribute(attribute_node, "Name") do
+          values = extract_attribute_values(attribute_node)
+          [{name, values}]
+        else
+          _ -> []
+        end
+      end)
+      |> Enum.into(%{})
+    else
+      _ -> %{}
+    end
+  end
+
+  defp extract_attribute_values(attribute_node) do
+    XmlNode.children(attribute_node, "*:AttributeValue")
```

Comment:
Ditto about supporting tags without prefix:

```suggestion
    XmlNode.children(attribute_node, ~r/.*:?AttributeValue$/)
```